main.tf

```diff
@@ -48,7 +66,12 @@ resource "aws_ec2_transit_gateway_vpc_attachment" "default" {
   appliance_mode_support = var.vpc_attachment_appliance_mode_support
   dns_support            = var.vpc_attachment_dns_support
   ipv6_support           = var.vpc_attachment_ipv6_support
-  tags                   = module.this.tags
+  tags = merge(
+    module.vpc_attach_label.tags,
+    {
+      "Name" = format("%s%s%s", module.vpc_attach_label.id, module.vpc_attach_label.delimiter, each.key)
```

Comment:
I don't see how much the `module.vpc_attach_label.id` is helping here... but I do see how `each.key` can be helpful here.